torch/distributions/truncated_normal.py

```diff
@@ -0,0 +1,100 @@
+import torch
+from torch.distributions.normal import Normal
+from torch.distributions.truncated_distribution import TruncatedDistribution
+import math
+
+
+def _standard_truncnorm_sample(lower_bound, upper_bound, sample_shape=torch.Size()):
+    r"""
+    Implements accept-reject algorithm for doubly truncated standard normal distribution.
+    (Section 2.2. Two-sided truncated normal distribution in [1])
+
+    [1] Robert, Christian P. "Simulation of truncated normal variables." Statistics and computing 5.2 (1995): 121-125.
+    Available online: https://arxiv.org/abs/0907.4010
+
+    Args:
+        lower_bound (Tensor): lower bound for standard normal distribution. Best to keep it greater than -4.0 for
+        stable results
+        upper_bound (Tensor): upper bound for standard normal distribution. Best to keep it smaller than 4.0 for
+        stable results
+    """
+    x = torch.randn(sample_shape)
+    done = torch.zeros(sample_shape).byte()
+    while not done.all():
```

Comment:
@suo should we wrap this function in a `@torch.jit.script` so the `while` loop is correctly recorded even when `TruncatedNormal.sample()` appears inside a `torch.jit.trace`? I have started using this pattern in Pyro whenever possible.

Reply:
@fritzo is there an example i can take a look at?

Reply:
https://github.com/pyro-ppl/pyro/blob/82c7cfd9a93ef06663f303a85885a59912aa5f47/pyro/distributions/hmm.py#L15